h2o-py/h2o/backend/connection.py

```diff
@@ -191,7 +191,42 @@ def url(self):
         return curl
 
 
-class H2OConnection(backwards_compatible()):
+@BackwardsCompatible(
+    class_attrs=dict(
+        __ENCODING__="utf-8",
+        __ENCODING_ERROR__="replace",
+        default=lambda: _deprecated_default(),
+        jar_paths=lambda: list(getattr(H2OLocalServer, "_jar_paths")()),
+        rest_version=lambda: 3,
+        https=lambda: __H2OCONN__.base_url.split(":")[0] == "https",
+        ip=lambda: __H2OCONN__.base_url.split(":")[1][2:],
+        port=lambda: __H2OCONN__.base_url.split(":")[2],
+        username=lambda: _deprecated_username(),
+        password=lambda: _deprecated_password(),
+        insecure=lambda: not getattr(__H2OCONN__, "_verify_ssl_cert"),
+        current_connection=lambda: __H2OCONN__,
+        check_conn=lambda: _deprecated_check_conn(),
+        make_url=lambda url_suffix, _rest_version=3: __H2OCONN__.make_url(url_suffix, _rest_version),
+        get=lambda url_suffix, **kwargs: _deprecated_get(__H2OCONN__, url_suffix, **kwargs),
+        post=lambda url_suffix, file_upload_info=None, **kwa: _deprecated_post(__H2OCONN__, url_suffix, file_upload_info=file_upload_info, **kwa),
+        delete=lambda url_suffix, **kwargs: _deprecated_delete(__H2OCONN__, url_suffix, **kwargs),
+        get_json=lambda url_suffix, **kwargs: _deprecated_get(__H2OCONN__, url_suffix, **kwargs),
+        post_json=lambda url_suffix, file_upload_info=None, **kwa: _deprecated_post(__H2OCONN__, url_suffix, file_upload_info=file_upload_info, **kwa),
+        rest_ctr=lambda: __H2OCONN__.requests_count,
+    ),
+    instance_attrs=dict(
+        cluster_is_up=lambda self: self.cluster.is_running(),
+        info=lambda self, refresh=True: self.cluster,
+        shutdown_server=lambda self, prompt=True: self.cluster.shutdown(prompt),
+        make_url=lambda self, url_suffix, _rest_version=3: "/".join([self._base_url, str(_rest_version), url_suffix]),
+        get=lambda *args, **kwargs: _deprecated_get(*args, **kwargs),
+        post=lambda *args, **kwargs: _deprecated_post(*args, **kwargs),
+        delete=lambda *args, **kwargs: _deprecated_delete(*args, **kwargs),
+        get_json=lambda *args, **kwargs: _deprecated_get(*args, **kwargs),
+        post_json=lambda *args, **kwargs: _deprecated_post(*args, **kwargs),
+    )
+)
+class H2OConnection(h2o_meta()):
```

Comment:
I was also considering simply deleting all those old backwards incompatibilities:
`These methods are deprecated since July 2016. Please remove them if it's 2017 already...`
But we're still using some of those internally....
So will create a separate ticket for this.

Reply:
thanks

Reply:
https://0xdata.atlassian.net/browse/PUBDEV-7114